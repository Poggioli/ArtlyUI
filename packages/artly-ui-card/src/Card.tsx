import { forwardRef } from "@artly-ui/core";
import { StyledCard } from "./styles";
import { CardProps } from "./types";

const Card = forwardRef<typeof StyledCard, CardProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledCard {...props} ref={forwardedRef} />;
  }
);

export { Card };
