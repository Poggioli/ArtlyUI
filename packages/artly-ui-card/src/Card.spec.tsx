import { render } from "@testing-library/react";
import { Card } from "./Card";

describe("Core component / Card", () => {
  it(`GIVEN a Card component
    WHEN render component
    SHOULD occurs render normaly`, () => {
    const rendered = render(<Card />);
    expect(rendered.container).toBeTruthy();
  });
});
