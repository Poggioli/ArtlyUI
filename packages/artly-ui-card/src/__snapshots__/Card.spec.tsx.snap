// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data display component / Card GIVEN a Card component
    WHEN render component
    SHOULD occurs render normaly 1`] = `
<div>
  <div
    class="artly-UI-c-bXbWpx artly-UI-c-bXbWpx-iiTKOFX-css artly-UI-c-uGsWN artly-UI-c-uGsWN-btuauh-shaded-true artly-UI-c-uGsWN-leqBWP-bordered-true"
  >
    <h3
      class="artly-UI-c-emYHtM artly-UI-c-emYHtM-jpIAgQ-level-5 artly-UI-c-emYHtM-iiBiFgP-css"
    >
      Card Title
    </h3>
    <h4
      class="artly-UI-c-chWhuJ artly-UI-c-chWhuJ-bzGfXE-variant-auxiliary artly-UI-c-chWhuJ-cmVlgk-align-left artly-UI-c-chWhuJ-ijlwcOl-css"
    >
      Card Subtitle
    </h4>
    <p
      class="artly-UI-c-chWhuJ artly-UI-c-chWhuJ-bkzKcy-variant-body2 artly-UI-c-chWhuJ-cmVlgk-align-left artly-UI-c-chWhuJ-ijlwcOl-css"
    >
      Card Description
    </p>
    <div
      class="artly-UI-c-bXbWpx artly-UI-c-bXbWpx-ibLIILq-css"
    >
      <button>
        Card Action
      </button>
    </div>
  </div>
</div>
`;
